projects/angular/src/data/datagrid/datagrid-row.spec.ts

```diff
@@ -543,6 +544,115 @@ export default function (): void {
         context.detectChanges();
       }
     });
+
+    describe('Details', function () {
+      let context: TestContext<ClrDatagrid<Item>, DatagridWithDisabledOrHiddenDetails>;
+
+      beforeEach(function () {
+        context = this.create(ClrDatagrid, DatagridWithDisabledOrHiddenDetails, DATAGRID_SPEC_PROVIDERS);
+        context.detectChanges();
+      });
+
+      it('displays a clickable double caret when the row have details', function () {
+        expect(context.clarityElement.querySelectorAll('button.datagrid-detail-caret-button').length).toBe(
+          context.testComponent.items.length
+        );
+      });
+
+      it('Detail button must contain aria-controls', function () {
+        const button = context.clarityElement.querySelector('button.datagrid-detail-caret-button');
+        expect(button).not.toBeNull();
+        expect(button.getAttribute('aria-controls')).not.toBeNull();
+      });
+
+      it('display the details when pre selected', fakeAsync(function () {
+        context.testComponent.preState = context.testComponent.items[0];
+        tick();
+        context.detectChanges();
+
+        const button = context.clarityElement.querySelector('button.datagrid-detail-caret-button');
+        const detailBody = document.querySelector('div.clr-dg-detail-body-wrapper') as HTMLElement;
+
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=left]')).not.toBeNull();
+        expect(detailBody.innerText).toBe('' + context.testComponent.preState.id);
+      }));
+
+      it('open and close the details', fakeAsync(function () {
+        const button = context.clarityElement.querySelector('button.datagrid-detail-caret-button');
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=right]')).not.toBeNull();
+
+        button.click();
+        tick();
+        context.detectChanges();
+
+        const detailBody = document.querySelector('div.clr-dg-detail-body-wrapper') as HTMLElement;
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=left]')).not.toBeNull();
+        expect(detailBody.innerText).toBe('' + context.testComponent.items[0].id);
+
+        button.click();
+        tick();
+        context.detectChanges();
+
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=right]')).not.toBeNull();
+      }));
```

Comment:
Not sure if we need that in the datagrid-row as this is already tested in detail unit tests and are part of the functionality there.

Reply:
It's not tested in `datagrid-detail.spec.ts`. It only tests detail pane, but not any open or close interaction through datagrid row.

Reply:
It's okey then, thanks!